plugin/src/main/scala/chisel3/internal/plugin/ChiselComponent.scala

```diff
@@ -232,7 +232,11 @@ class ChiselComponent(val global: Global, arguments: ChiselPluginArguments)
               val onames: List[Option[String]] =
                 fieldsOfInterest.zip(names).map { case (ok, name) => if (ok) Some(name) else None }
               val newRHS = transform(rhs)
-              val named = q"chisel3.internal.plugin.autoNameRecursivelyProduct($onames)($newRHS)"
+              // for tuple destructuring assignments, prefix using the concatenation of all names
+              // e.g. for `val (a, b, c) = ...` use prefix "a_b_c"
```

Comment:
why would designers want the prefix on all of them? Usually this tupling is just done for convenience that a code block is returning 3 things, wouldn't we want to prefix each with just its corresponding variable name?

Reply:
The returned things aren't prefixed, but any local values in the RHS callstack are, for example:
```scala
val (a, b) = {
  val x, y = Wire(UInt(3.W))
  val sum = WireInit(x + y)
  (x, y)
}
```
This will generate (simplified a bit):
```
wire a : UInt<3>
wire b : UInt<3>
wire a_b_sum : UInt<3>
a_b_sum <= tail(add(a, b,), 1)
```
Whereas currently it generates
```
wire a : UInt<3>
wire b : UInt<3>
wire sum : UInt<3>
sum <= tail(add(a, b,), 1)
```

Reply:
I'm not certain if this is a change we really want, or if we do want to add prefixes here, what they should be, the 4 options are:
1. No prefix, the user can call prefix if they want (current behavior)
2. Pick the first val as a prefix (`a`)
3. Pick the last val as a prefix (`b`)
4. Use all vals as prefixes (`a_b` would be the result)

This PR is (4).

Reply:
IMHO the only thing that makes sense here is 1